CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+# Developer Guidelines
+
+## Code Style
+
+### Existing non-compliant code
+The code that does not comply with these guidelines will be fixed in separate pull request(s). Any new contribution should follow closely these guidelines but should not change style in the existing GridKit code.
+
+### Error handling
+Return values of member functions should be of type `int` and used for error handling. Functions return 0 if no error is encounter, return positive value for warnings and recoverable error, and negative value for irrecoverable errors.
```

Comment:
This seems a little unwieldy - does this mean any outputs must be passed by reference or pointer? Perhaps we could consider returning `std::optional` for warnings/recoverable errors and throwing for unrecoverable errors?

Reply:
Outputs are indeed passed by reference or a pointer in the current code. Even if the output is the function return value it would still be a pointer to an object such as Jacobian matrix. I don't see a major disadvantage if we reserve return value for function execution status.

Having said that, what you suggest is a good idea, so please create a separate issue with more details. That would require some refactoring of the GridKit code, as well.